test/e2e/validation_test.go

```diff
@@ -503,3 +503,39 @@ func (tc *testContext) validateUpgradeableCondition(expected meta.ConditionStatu
 	}
 	return nil
 }
+
+func (tc *testContext) testClusterDNSForwardQuery(t *testing.T) {
+	// test DNS Server with forward query with a 60s timeout
+	winJob, err := tc.createWindowsServerJob("win-dns-tester-forward-query",
+		"ping -l 0 -n 1 -w 60000 www.openshift.com",
+		nil)
+	require.NoError(t, err, "could not create Windows tester job")
+	// cleanup
+	defer tc.deleteJob(winJob.Name)
+	// assert
+	err = tc.waitUntilJobSucceeds(winJob.Name)
+	assert.NoError(t, err, "Windows tester job failed")
+}
+
+func (tc *testContext) testClusterDNSInvalidRecord(t *testing.T) {
+	// test DNS Server with invalid record
+	winJob, err := tc.createWindowsServerJob("win-dns-tester-invalid-record",
+		"ping -l 0 -n 1 www.invalid--dns--record.test",
+		nil)
+	require.NoError(t, err, "could not create Windows tester job")
+	// cleanup
+	defer tc.deleteJob(winJob.Name)
+	// assert
+	err = tc.waitUntilJobSucceeds(winJob.Name)
+	assert.Error(t, err, "Windows tester job expected to fail")
+}
+
+// testClusterDNS test the DNS server in a Windows pod
+func testClusterDNS(t *testing.T) {
+	// create context
+	testCtx, err := NewTestContext()
+	require.NoError(t, err)
+	// run tests
+	t.Run("Forward Query", testCtx.testClusterDNSForwardQuery)
+	t.Run("Invalid Record", testCtx.testClusterDNSInvalidRecord)
+}
```

Comment:
Ideally this needs to be right after the other existing tests in this file but this file is a mess and calls for some reorg and cleanup. I suggest moving this to next best possible space, i.e. after `testNodeIPArg`. The intent is to collect all the test we run at the start.

Reply:
Makes sense.